FirstTask

RichardVas opened the pull request:
@hegyhati

RichardVas committed:
Add files via upload

vizvezetek reviewed (commented):
@hegyhati Mi a sorsa a "Add files via upload" feltöltéseknek?

vizvezetek commented on Fighter.h:
```diff
@@ -0,0 +1,58 @@
+	}
+	int getHP() { return this->HP; }
+	int getDMG() const { return this->DMG; }
+	void setHP(int x) { HP = x; }
```
Setter függvény nélkül oldjuk meg a feladatot.

vizvezetek commented on Fighter.h:
```diff
@@ -0,0 +1,58 @@
+	int getDMG() const { return this->DMG; }
+	void setHP(int x) { HP = x; }
+	std::string getname() const { return this->name; }
+	void print_character()
+	{
+		std::cout << name << "\t" << "HP: " << getHP() << "  " << "Dmg: " << getDMG() << std::endl;
+	}
```
Egy alap működést megvalósító osztályban ne használjunk cout-ot.

vizvezetek commented on Fighter.h:
```diff
@@ -0,0 +1,58 @@
+
+public:
+
+	Fighter(std::string iname, int ihp, int idmg) : name(iname), HP(ihp), DMG(idmg) {}
```
Egy destruktort még elbírna az osztály.

vizvezetek commented on Fighter.h:
```diff
@@ -0,0 +1,58 @@
+	{
+		std::cout << name << "\t" << "HP: " << getHP() << "  " << "Dmg: " << getDMG() << std::endl;
+	}
+	void duel(Fighter *enemy){
+		bool can_attack = true;
+		while (this->getHP() != 0 && enemy->getHP() != 0)
+		{
+			this->print_character();
+			enemy->print_character();
+			if (can_attack) {
+				std::cout << this->getname() << "--->" << enemy->getname() << std::endl;
+				enemy->take_dmg(this->getDMG());
+				if (enemy->getHP() < 0)
+					enemy->setHP(0);
+				can_attack = false;
+			}
+			else
+			{
+				std::cout << enemy->getname() << "--->" << this->getname() << std::endl;
+				this->take_dmg(enemy->getDMG());
+				if (getHP() < 0)
+					setHP(0);
+				can_attack = true;
+			}
+		}
+		this->print_character();
+		enemy->print_character();
+		if (this->getHP() > enemy->getHP())
+		{
+			std::cout <<enemy->getname()<<" died. "<< this->getname() << " wins.\n";
+		}
+		else	
+		{
+			std::cout << this->getname() << " died. " << enemy->getname() << " wins.\n";
+		}
+	}
```
Ha ez kikerül a classon kívülre, akkor maradhat a cout. Az osztályon belül más alternatívát kell használni.

vizvezetek reviewed (changes requested):
Nos. Első és legfőbb probléma hogy "Add files via upload"-ot nem szeretnék meglátni. Ez nem az a feladat.
A kóddal kapcsolatban vannak kisebb-nagyobb problémák:
A fighter osztály csak alap adattagokat és metódusokat tartalmazzon. Maga a játék "szimulációt" szedjétek ki onnan, nem ott kell lennie. 
A Fighter.h-t szedjétek szét Fighter.h-ra és Fighter.cpp-re. A jelenlegi állapotában nem túl elegáns.

hegyhati commented:
@RichardVas @vizvezetek elsore figyelmeztetes, masodjara reject.

RichardVas committed:
Fix header format

Added a destructor, overloaded the << operator, updated the basic methods and removed the duel method as requested.

RichardVas committed:
Implement the methods declared in Fighter.h

RichardVas committed:
Move duel from class

Implement a clash between two units, using the duel(Fighter attacker, Fighter defender) function.

RichardVas committed:
Add job description

Jelenlegi feladat állapot rövid megfogalmazása.

RichardVas commented:
@vizvezetek , @hegyhati 
A kért változtatásokat elvégeztük.

vizvezetek reviewed (approved):
@hegyhati Részemről rendben.
